### PR TITLE
Fixing Patch logic for Tokens

### PR DESCRIPTION
**Pull request by danielyoo227**
Fixing PATCH logic. Also added reload logic for the GET flow since we need IdTokenInfo's information to map some of the TokenDTO data.

### CODE REVIEW

File: 00_Base/src/repository/TokensRepository.ts [hunk 1]
```diff
@@ -285,15 +312,79 @@ export class TokensRepository extends SequelizeRepository<OcpiToken> {
...
+      transaction,
+    });
+  }
+
+  private mapTokenDTOToPartialAuthorization(
+    existingAuth: Authorization,
+    tokenDTO: Partial<TokenDTO>,
+  ): Partial<IdTokenInfoType> {
+    const idTokenInfo: IdTokenInfoType = {
+      status: existingAuth.idTokenInfo?.status!,
```
Comment (yuvenus) [inline 00_Base/src/repository/TokensRepository.ts:365]: linter error her, seems that it doesn't like the not null assertion here 🤔 i think it's solvable by making the type of id token into a `Partial<IdTokenInfoType>` instead...?

**Review by yuvenus — APPROVED** [review]
approved! seems there's a linter error that needs to be resolved before this is mergeable though 🤔

File: 00_Base/src/repository/TokensRepository.ts [hunk 2]
```diff
@@ -285,15 +312,79 @@ export class TokensRepository extends SequelizeRepository<OcpiToken> {
...
+      });
+    }
+
+    return await existingAuth.reload({
+      include: [{ model: IdTokenInfo }],
+      transaction,
+    });
+  }
+
+  private mapTokenDTOToPartialAuthorization(
```
Comment (langiole) [inline 00_Base/src/repository/TokensRepository.ts:360]: Maybe we can put this method in the tokens mapper file?

**Review by langiole — APPROVED** [review]

File: 00_Base/src/repository/TokensRepository.ts (hunk 2 above)
Comment (danielyoo227) [reply 00_Base/src/repository/TokensRepository.ts]: Yup, makes sense. Moved!

**Pull request merged**